src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,290 @@
+import pandas as pd
+
+from viadot.utils import APIError, handle_api_response
+
+from ..signals import SKIP
+from ..utils import add_viadot_metadata_columns, validate
+from .base import Source
+
+
+class Eurostat(Source):
+    """
+    Class for creating instance of Eurostat connector to REST API by HTTPS response (no credentials required).
+    """
+
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        """
+        It is using HTTPS REST request to pull the data.
+        No API registration or API key are required.
+        Data will pull based on parameters provided in dynamic part of the url.
+        Example of url: https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/TEIBS020/?format=JSON&lang=EN&indic=BS-CSMCI-BAL
+        Static part: https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data
+        Dynamic part: /TEIBS020/?format=JSON&lang=EN&indic=BS-CSMCI-BAL
+        Please note that for one dataset there are usually multiple data regarding different subjects.
+        In order to retrive data that you are interested in you have to provide parameters with codes into 'params'.
+        """
+
+        super().__init__(*args, **kwargs)
+
+    def get_parameters_codes(self) -> dict:
+        """Function for getting available parameters with codes from dataset.
+        Raises:
+            ValueError: If the response from the API is empty or invalid.
+        Returns:
+            Dict: Key is parameter and value is a list of available codes for specific parameter.
+        """
+
+        try:
+            response = handle_api_response(self.base_url)
+            data = response.json()
+        except APIError:
+            self.logger.error(
+                f"Failed to fetch data for {self.dataset_code}, please check correctness of dataset code!"
+            )
+            raise ValueError("DataFrame is empty!")
+
+        # getting list of available parameters
+        available_params = data["id"]
+
+        # dictionary from JSON with keys and reletad codes values
+        dimension = data["dimension"]
+
+        # Assigning list of available codes to specific parameters
+        params_and_codes = {}
+        for key in available_params:
+            if key in dimension:
+                codes = list(dimension[key]["category"]["index"].keys())
+                params_and_codes[key] = codes
+        return params_and_codes
+
+    def make_params_validation(self):
+        """Function for validation of given parameters in comparison
+        to parameteres and their codes from JSON.
```

Comment:
1. Docstrings should be in imperative (this applies to all the other docstrings as well but I'm not going to repeat this comment everywhere to not pollute)
2. Sorry, I didn't understand anything more than "this validates parameters". I'd say it's best to either be brief (like below suggestion) or provide a summary of your validation logic (list things you're validating).

```suggestion
        """Validate API request parameters."""
```

Reply:
Changed.

Reply:
Thanks. Can you pls also apply this rule to the rest of them, as per point 1.?